src/web/app/pages-student/student-page.component.ts

```diff
@@ -29,7 +29,11 @@ export class StudentPageComponent implements OnInit {
       if (res.logoutUrl) {
         this.logoutUrl = `${this.backendUrl}${res.logoutUrl}`;
       }
-      this.isValidUser = res.user && res.user.isStudent;
+      if (res.user) {
+        this.isValidUser = res.user.isStudent;
+      } else {
+        window.location.href = res.studentLoginUrl;
```

Comment:
Should use Angular service `$location`?

Reply:
`$location` in an AngularJS thing, not Angular. I however will check if the [Location API](https://angular.io/api/common/Location) can be used.

Reply:
Tried; Location API is not effective for navigating to absolute URLs.

Reply:
Not sure if this will become a troublesome dependency when writing unit test.